packages/vite/src/plugins/dev-ssr-css.ts

```diff
@@ -22,9 +22,22 @@ export function devStyleSSRPlugin (options: DevStyleSSRPluginOptions): Plugin {
         moduleId = moduleId.slice(options.srcDir.length)
       }
 
-      // When dev `<style>` is injected, remove the `<link>` styles from manifest
-      const selectors = [joinURL(options.buildAssetsURL, moduleId), joinURL(options.buildAssetsURL, '@fs', moduleId)]
-      return code + selectors.map(selector => `\ndocument.querySelectorAll(\`link[href="${selector}"]\`).forEach(i=>i.remove())`).join('')
+      // When dev <style> is injected, remove the <link> styles from manifest
+      const selectors = [
+        joinURL(options.buildAssetsURL, moduleId),
+        joinURL(options.buildAssetsURL, '@fs', moduleId),
+      ]
+
+      // Added custom attribute check
+      const updatedCode = selectors.map(selector => `
+        document.querySelectorAll('link[href="${selector}"]').forEach(i => {
+          if (i.getAttribute('data-vite-dev-style')) {
```

Comment:
what is going to add this attribute? currently styles look like this:

```html
<!DOCTYPE html><html  data-capo=""><head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="/_nuxt/assets/test.css">
<link rel="stylesheet" href="/_nuxt/app.vue?vue&type=style&index=0&scoped=938b83b0&lang.css">
<script type="module" src="/_nuxt/@vite/client" crossorigin></script>
<!-- ... -->
```

Reply:
so i wasn't sure the best way to implement this:

started a gist [https://gist.github.com/izzygld/a193409ad84562df75abe4cc26cc5602](https://gist.github.com/izzygld/a193409ad84562df75abe4cc26cc5602 )

Reply:
I'm not entirely tracking. How would this solve the linked issue?

Looking into the issue, it seems maybe we need to _render_ the style links differently - adding the `layer` attribute.

One note: `transformIndexHtml` isn't called with frameworks like Nuxt that consume Vite.